.github/workflows/pr.yml

```diff
@@ -39,7 +39,7 @@ jobs:
 
     name: Unit Tests
 
-    runs-on: macos-14-xlarge
+    runs-on: macos-15
```

Comment:
This is an optional change to move to the free runner, which adds about ~3 minutes of extra test runtime. I'm happy to change this back to `macos-15-xlarge` if requested.